dashboard/app/models/datablock_storage_kvp.rb

```diff
@@ -37,7 +36,9 @@ def self.get_kvps(project_id)
 
   def self.set_kvps(project_id, key_value_hashmap, upsert: true)
     kvps = key_value_hashmap.map do |key, value|
-      {project_id: project_id, key: key, value: value.to_json}
+      kvp_attr = {project_id: project_id, key: key, value: value.to_json}
+      DatablockStorageKvp.new(kvp_attr).valid?
```

Comment:
OK so because of these above
```rb
validate :max_kvp_count, on: :create
validate :max_value_length, on: [:create, :update]
```
the new is a create and those 2 functions get called with the kvp_attr passed in as `value`

magic!!